### PR TITLE
add query params to entity-management route

### LINKED ISSUE #2657: Painful paginated entity navigation
* **Overview of the issue**
When an entity has "pagination" pagination and you view an item, clicking on "back" sends you to the first page instead of the page you were on before viewing the item

* **Motivation for or Use Case** 
This is really painful when you want to see item details one item by one item of a very long list.

* **JHipster Version(s)** 
2.26.2

* **JHipster configuration, a `.yo-rc.json` file generated in the root folder** 
```json
{
  "generator-jhipster": {
    "baseName": "jhipsterstandard",
    "packageName": "com.mycompany.myapp",
    "packageFolder": "com/mycompany/myapp",
    "authenticationType": "session",
    "hibernateCache": "ehcache",
    "clusteredHttpSession": "no",
    "websocket": "no",
    "databaseType": "sql",
    "devDatabaseType": "h2Memory",
    "prodDatabaseType": "postgresql",
    "searchEngine": "no",
    "useSass": false,
    "buildTool": "maven",
    "frontendBuilder": "grunt",
    "enableTranslation": true,
    "enableSocialSignIn": false,
    "rememberMeKey": "5e77ac8064a0afdbff55ed4240fd6493ffffffff",
    "testFrameworks": [
      "gatling"
    ]
  }
```

* **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory** 
Foo.json
```json
{
    "relationships": [],
    "fields": [
        {
            "fieldId": 1,
            "fieldName": "foo",
            "fieldType": "String"
        }
    ],
    "changelogDate": "20160119161144",
    "dto": "no",
    "service": "no",
    "pagination": "pagination"
}
```
* **Browsers and Operating System** 
Firefox 42, probably all browsers

* **Reproduce the error** 
* Connect to h2 console
* Create some foo items
```sql
INSERT INTO FOO(foo) VALUES ('foo'),('foo'),('foo'),('foo'),('foo'),('foo'),('foo'),('foo'),('foo'),('foo'),('foo'),('foo'),('foo'),('foo'),('foo'),('foo'),('foo'),('foo'),('foo'),('foo'),('foo'),('foo');
```
* Go to Entities -> Foo
* Go to page 2
* Select id 22
* Click on "back" --> the item list is on page 1 whereas the user would want to be on page 2

### PR DESCRIPTION
**Pull request by geraldhumphries**
add query params to entity-management route when pagination/pager is enabled

fix #2657

### CODE REVIEW

**Comment by geraldhumphries** [comment]
This pull request does NOT include the fix for #2387. That will come after this is merged.

**Comment by deepu105** [comment]
@gzsombor @gmarziou what do you think of this?

**Comment by geraldhumphries** [comment]
Yeah I have some concerns over my code:

1. Should I split the `sort` url param up into `predicate` (or leave it as `sort` to be more user friendly) and `direction`?
1. Should I rename `transition()` to something else?
1. The sort-directive change was a quick-fix, but it works well enough. Have any suggestions for how I can improve it?

**Comment by deepu105** [comment]
Im not sure as well @jhipster/developers can someone take a second look as well

**Comment by jdubois** [comment]
I don't have much opinion here, it looks OK to me.

File: generators/entity/templates/src/main/webapp/app/entities/_entity-management.controller.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 'use strict';
 
 angular.module('<%=angularAppName%>')
-    .controller('<%= entityClass %>ManagementController', function ($scope, $state<% if (fieldsContainBlob) { %>, DataUtils<% } %>, <%= entityClass %><% if (searchEngine == 'elasticsearch') { %>, <%= entityClass %>Search<% } %><% if (pagination != 'no') { %>, ParseLinks<% } %> <%_ if (pagination == 'pager' || pagination == 'pagination'){ %>, paginationConstants<% } %>) {
+    .controller('<%= entityClass %>ManagementController', function ($scope, $state<% if (fieldsContainBlob) { %>, DataUtils<% } %>, <%= entityClass %><% if (searchEngine == 'elasticsearch') { %>, <%= entityClass %>Search<% } %><% if (pagination != 'no') { %>, ParseLinks<% } %> <%_ if (pagination == 'pager' || pagination == 'pagination'){ %>, AlertService, pagingParams, paginationConstants<% } %>) {
```
Comment (deepu105) [inline generators/entity/templates/src/main/webapp/app/entities/_entity-management.controller.js]: Where is pagingparam and alerservice used? If not plz remove the injects

File: generators/entity/templates/src/main/webapp/app/entities/_entity-management.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ angular.module('<%=angularAppName%>')
         $stateProvider
             .state('<%= entityStateName %>', {
                 parent: 'entity',
-                url: '/<%= entityUrl %>',
+                url: '/<%= entityUrl %><% if (pagination == 'pagination' || pagination == 'pager') { %>?page&sort&search<% } %>',
```
Comment (deepu105) [inline generators/entity/templates/src/main/webapp/app/entities/_entity-management.js]: The url syntax looks weird is it standard?

File: generators/entity/templates/src/main/webapp/app/entities/_entity-management.js [hunk 3]
```diff
@@ -16,7 +16,31 @@ angular.module('<%=angularAppName%>')
                         controller: '<%= entityClass %>ManagementController'
                     }
                 },
-                resolve: {<% if (enableTranslation){ %>
+                <%_ if (pagination == 'pagination' || pagination == 'pager'){ _%>
+                params: {
```
Comment (deepu105) [inline generators/entity/templates/src/main/webapp/app/entities/_entity-management.js]: What is this used for?

File: generators/entity/templates/src/main/webapp/app/entities/pagination-template.ejs [hunk 4]
```diff
@@ -1,23 +1,30 @@
...
+            var onSuccess = function (data, headers) {
                 $scope.links = ParseLinks.parse(headers('link'));
                 $scope.totalItems = headers('X-Total-Count');
-                $scope.<%= entityInstancePlural %> = result;
-            });
+                $scope.<%= entityInstancePlural %> = data;
+                $scope.page = pagingParams.page;
+            };
+            var onError = function (error) {
+                AlertService.error(error.data.message);
```
Comment (deepu105) [inline generators/entity/templates/src/main/webapp/app/entities/pagination-template.ejs]: How is i18n handled for the error message

File: generators/entity/templates/src/main/webapp/app/entities/_entity-management.controller.js (hunk 1 above)
Comment (deepu105) [inline generators/entity/templates/src/main/webapp/app/entities/_entity-management.controller.js]: Ok found it, ignore above comment

File: generators/entity/templates/src/main/webapp/app/entities/_entity-management.js (hunk 2 above)
Comment (geraldhumphries) [inline generators/entity/templates/src/main/webapp/app/entities/_entity-management.js]: Yes: https://github.com/angular-ui/ui-router/wiki/URL-Routing#query-parameters

File: generators/entity/templates/src/main/webapp/app/entities/_entity-management.js (hunk 3 above)
Comment (geraldhumphries) [inline generators/entity/templates/src/main/webapp/app/entities/_entity-management.js]: These are used to set the defaults for each param and guarantee they aren't undefined. For squash, it means that the param won't show in the url if it is set to the default - so if page is 1 you won't see `page=1`.

Comment (geraldhumphries) [inline generators/entity/templates/src/main/webapp/app/entities/_entity-management.js]: Perhaps `search: null` is unnecessary.

File: generators/entity/templates/src/main/webapp/app/entities/pagination-template.ejs (hunk 4 above)
Comment (geraldhumphries) [inline generators/entity/templates/src/main/webapp/app/entities/pagination-template.ejs]: Good point, i18n is not taken into account. It just prints the string from the server. Do you have a suggestion to improve it?

There was no error handling before and this is better than nothing IMO.

Comment (deepu105) [inline generators/entity/templates/src/main/webapp/app/entities/pagination-template.ejs]: Ok lets merge this and take acre of i18n

Check how validation messages are translated to get an idea

**Pull request merged**